File: Leetcode/334/334.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int n = (int)nums.size();
        if (n < 3) {
            return false;
        }
        vector<int> minFromLeft(n);
        minFromLeft[0] = nums[0];
        for (int i = 1; i < n; ++i) {
            minFromLeft[i] = min(nums[i],minFromLeft[i-1]);
        }
        vector<int> maxFromRight(n);
        maxFromRight[n-1] = nums[n-1];
        for (int i = n-2; i >= 0; --i) {
            maxFromRight[i] = max(nums[i],maxFromRight[i+1]);
        }
        for (int i = 1; i <= n-2; ++i) {
            if (minFromLeft[i-1] < nums[i] && nums[i] < maxFromRight[i+1]) {
                return true;
            }
        }
        return false;
    }
};

File: Leetcode/334/334.go
func max(a, b int) int {
	if a < b {
		return b
	}
	return a
}
func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func increasingTriplet(nums []int) bool {
	if len(nums) <= 2 {
		return false
	}
	maxFromRight := make([]int, len(nums))
	maxFromRight[len(nums)-1] = nums[len(nums)-1]
	for i := len(nums)-2; i >= 0; i-- {
		maxFromRight[i] = max(maxFromRight[i+1], nums[i])
	}
	minFromLeft := nums[0]
	for i := 1; i < len(nums)-1; i++ {
		if minFromLeft < nums[i] && nums[i] < maxFromRight[i+1] {
			return true
		}
		minFromLeft = min(minFromLeft, nums[i])
	}
	return false
}